packages/block-editor/src/hooks/block-style-variation.js

```diff
@@ -194,9 +256,14 @@ function useBlockStyleVariation( name, variation, clientId ) {
 	}, [] );
 
 	return useMemo( () => {
-		const styles = mergedConfig?.styles ?? globalStyles;
-		const variationStyles =
-			styles?.blocks?.[ name ]?.variations?.[ variation ];
+		const variationStyles = getVariationStylesWithRefValues(
+			{
+				settings: mergedConfig?.settings ?? globalSettings,
+				styles: mergedConfig?.styles ?? globalStyles,
+			},
```

Comment:
Creating a "complete" theme.json object seemed the easiest approach to reason about when `ref` values could target any value within `settings` or `styles` etc.